src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -255,6 +255,20 @@ public bool ContainsAnalyzerConfigDocument(DocumentId documentId)
             return await GetSourceGeneratedDocumentAsync(documentId, cancellationToken).ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// Gets a document, additional document, analyzer config document or a source generated document in this solution with the specified document ID.
+        /// </summary>
+        internal async ValueTask<TextDocument?> GetTextDocumentAsync(DocumentId documentId, CancellationToken cancellationToken = default)
+        {
```

Comment:
Given that this does so much more than `GetDocumentAsync` I wonder if the name could make that clearer. `GetAnyTextDocumentAsync` perhaps?

I don't feel too strongly though, the doc comment is certainly good.

Reply:
The semantics of "any" is already in "text" based on our naming of the document classes.

Reply:
@jasonmalinowski Opinions?

Reply:
I hate this as much as I hate the other naming we have here. :smile: But it's consistent and at least immediately obvious (to me!) what this is doing given the existing naming, so I'm OK with it.

Reply:
If that's what "text" means, then thats fine by me. Thanks for the explanation

Reply:
Yes, because `TextDocument` is a common base class for `Document`, `AdditionalDocument`, `AnalyzerConfigDocument` and `SourceGeneratedDocument`